### DISCUSSION TITLE
How reliable is Serverpod for real?

### DISCUSSION THREAD

**Discussion by mikeamir** [serverpod/serverpod#2355] [General]
Hello, I have a few questions about Serverpod.

1. Are there any real-life apps that use Serverpod in production?

2. Are there any benchmarks that show whether Serverpod has the required potential to compete with servers made using other popular backend options like NodeJS, Laravel, etc.?

3. Does ServerPod have any common types of APIs, like REST APIs? because sometimes we use different technologies other than Flutter for certain platforms, and not being able to integrate the Serverpod backend with these technologies is a huge limitation. This way, we must wrap the generated client with an extra Dart REST API. That alone makes Serverpod not a realistic option for a large-scale application.

4. If I had two applications, one of which is a client app and one of which is a dashboard, could I generate more than one client where I could control which endpoints were generated in each client?

+ A complaint:

The authentication tutorials are so focused on the widgets that Serverpod provides out of the box which is so weird.

**Comment by Isakdl** [comment]
Hey @mikeamir 

You got some good questions! 

1. One notable example I can give you right now is MyOpNotes: https://www.myopnotes.co.uk/. I know of many more but I am unsure which I may disclose or not hence ill leave that out for now.  With that said I think creating a page with case studies/example products would be a good idea.
2. Yes, we had a very helpful member of the community who did a benchmark against other platforms, see: https://github.com/serverpod/serverpod/discussions/1791
3. We have the relic webserver https://docs.serverpod.dev/concepts/webserver, making it possible to create a REST api. It is experimental but we have plans to improve it, and I agree it is an important feature.
4. Today you cannot generate two different clients from the same backend project, if this is something you think makes sense please feel free to file an issue and explain how and why! But remember that dart has tree-shaking and will remove any unused code at compile time, meaning that in practice you will only ship the client methods you actually use.

Fair complaint about the authentication tutorials, they were created as an easy-to-get-started path. But you can of course create your own custom UI if you like! We have some documentation here on how to interact with the auth client: https://docs.serverpod.dev/concepts/authentication/providers/email#custom-ui-with-emailauthcontroller

**Reply by mikeamir** [reply]
Thanks a lot for your time.
